pkg/kubelet/apis/podresources/server_v1.go

```diff
@@ -70,3 +70,33 @@ func (p *v1PodResourcesServer) List(ctx context.Context, req *v1.ListPodResource
 		PodResources: podResources,
 	}, nil
 }
+
+// GetAllocatableResources returns information about all the devices known by the server
+func (p *v1PodResourcesServer) GetAllocatableResources(ctx context.Context, req *v1.AllocatableResourcesRequest) (*v1.AllocatableResourcesResponse, error) {
+	metrics.PodResourcesEndpointRequestsTotalCount.WithLabelValues("v1").Inc()
+
+	allDevices := p.devicesProvider.GetAllDevices()
+	var respDevs []*v1.ContainerDevices
+
+	for resourceName, resourceDevs := range allDevices {
+		for devID, dev := range resourceDevs {
+			for _, node := range dev.GetTopology().GetNodes() {
```

Comment:
func (m* Device) GetTopology gould return nil, if device plugin doesn't fill it, as we know now some device plugin in case of non NUMA architecture .is doing it

Reply:
Thanks for the reminder! Using the Get*() accessor makes us safe from this case, and I added this https://github.com/kubernetes/kubernetes/pull/95734/commits/cd06adb00b30bf7dd9ce115b3c07b112b94ec735 to test this occurrence.

Reply:
But if we don't have topology, but have devices, do we still need to return it? For our use case not, but maybe someone else doesn't require topology... hmm in this case it's better to rely on v1.Node...Allocatable.

Reply:
True. Let me review what List() does in this case, we need to be consistent (or to clearly document why we are not consistent...)

Reply:
Decided to report devices without topology information, let the client decide.